### ISSUE TITLE
$name should respect SCOUT_PREFIX

### ISSUE THREAD

**Issue by naden** [babenkoivan/scout-elasticsearch-driver#69]
In `IndexConfigurator` the method `getName` should prefix the default name generation with `SCOUT_PREFIX` from the .env file. This comes in handy, if you run multiple L5 apps with a single ES instance.

**Comment by iget-master** [comment]
This makes sense:

I run both production and staging environment under same ES instance, and now we are changing some of our ES indexes and want to deploy to staging, but it will affect production.

**Comment by iget-master** [comment]
FYI: I've solved this issue by extending the IndexConfigurator class:

```
class BaseIndexConfigurator extends IndexConfigurator {
    public function getName()
    {
        $name = parent::getName();
        $env = App::environment();

        if ($env !== 'production') {
            $name = "{$env}_{$name}";
        }

        return $name;
    }
}
```

Not the ideal solution, but solves for me.

**Comment by babenkoivan** [comment]
Hi guys. You are absolutely right, this can be considered as a bug. I fixed the issue in `v3.5.0`.
